pages/article/ArticleHandler.inc.php

```diff
@@ -55,25 +67,61 @@ function authorize($request, &$args, $roleAssignments) {
 	 */
 	function initialize($request, $args) {
 		$articleId = isset($args[0]) ? $args[0] : 0;
-		$galleyId = isset($args[1]) ? $args[1] : 0;
 
-		$journal = $request->getContext();
-		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
-		$publishedArticle = $publishedArticleDao->getPublishedArticleByBestArticleId((int) $journal->getId(), $articleId, true);
+		$this->journal = $request->getContext();
 
 		$issueDao = DAORegistry::getDAO('IssueDAO');
+		$articleDao = DAORegistry::getDAO('ArticleDAO');
+		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
+
+		// get published article object (handle pub ids)
+		$publishedArticle = $publishedArticleDao->getPublishedArticleByBestArticleId((int) $this->journal->getId(), $articleId, false, null);
+
+		// get all published previous article versions
+		$this->previousRevisions = $publishedArticleDao->getPublishedSubmissionRevisions($publishedArticle->getId(), $this->journal->getId(), SORT_DIRECTION_DESC);
+
+		// get the most recent article version
+		$this->latestSubmissionRevision = $publishedArticleDao->getLatestRevisionId($publishedArticle->getId(), $this->journal->getId());
+
+		// set latest submission revision as default
+		$this->submissionRevision = $this->latestSubmissionRevision;
+
+		// get metadata of the latest version
+		$publishedArticle = $publishedArticleDao->getPublishedArticleByBestArticleId((int) $this->journal->getId(), $articleId, false, $this->latestSubmissionRevision);
+
+		// get data of publishedArticle
 		if (isset($publishedArticle)) {
 			$issue = $issueDao->getById($publishedArticle->getIssueId(), $publishedArticle->getJournalId(), true);
 			$this->issue = $issue;
 			$this->article = $publishedArticle;
 		} else {
 			$articleDao = DAORegistry::getDAO('ArticleDAO');
-			$article = $articleDao->getById((int) $articleId, $journal->getId(), true);
+			$article = $articleDao->getById((int) $articleId, $this->journal->getId(), true);
```

Comment:
hm, I did not see a such function in ArticleDao. Also this function is only used for unpublished articles, so the articleId can only be an int, I guess. 🐰